### DISCUSSION TITLE
TS error when specifying store options

### DISCUSSION THREAD

**Discussion by danvratil** [iendeavor/pinia-plugin-persistedstate-2#14] [Q&A] [answered]
Hi, this project looks awesome, but I'm struggling to get it working with my code.

I want to specify store-specific options, so I followed the example in README:

```
export const useUiStateStore = defineStore(
  'uiState',
  () => {
    const elementOpened = ref(false)


    return {
      elementOpened,
      toggleElement: () => {
        elementOpened.value = !elementOpened.value
        return elementOpened.value
      }
    }
  },
  {
    persistedState: {
      persist: true
    }
  }
)
```

Unfortunately, this yields an error:
```
TS2345: Argument of type '{ persistedState: { persist: boolean; }; }' is not assignable to parameter of type 'DefineSetupStoreOptions<"uiState", _UnwrapAll<Pick<{ elementOpened: Ref<boolean>; toggleElement: () => boolean; }, "elementOpened">>, _UnwrapAll<Pick<{ elemen
tOpened: Ref<boolean>; toggleElement: () => boolean; }, never>>, Pick<...>>'.
  Object literal may only specify known properties, and 'persistedState' does not exist in type 'DefineSetupStoreOptions<"uiState", _UnwrapAll<Pick<{ elementOpened: Ref<boolean>; toggleElement: () => boolean; }, "elementOpened">>, _UnwrapAll<Pick<{ elementOpened: Ref<boolea
n>; toggleElement: () => boolean; }, never>>, Pick<...>>'.

```

I'm sure I'm doing something wrong on my side, but I can't seem to figure out what it is :-) Any ideas, please?

**Comment by soc221b** [comment]
@danvratil 

I cannot reproduce this 😞 , can you provide a minimal reproduction. (You can use [`vue3-example`](https://codesandbox.io/s/github/iendeavor/pinia-plugin-persistedstate-2/tree/main/examples/vue3-example) template on codesandbox).

We need to know something like which `pinia`, `pinia-plugin-persistedstate-2` version you use.

Maybe you can try to upgrade to latest version: `pinia-plugin-persistedstate-2@^1.0.0`.

**Comment by danvratil** [comment, marked as answer]
OK, I think I figured it out while experimenting with the codesandbox.

My setup is slightly specific - I have a monorepo with several subprojects -  one of them is a library that defines the store, another one is a Vue app that actually takes care of creating Pinia and instantiating the store. I had `pinia-plugin-persistedstate-2` installed in both the library and the app, but I did not import it in the library, just in the app. Thus, the `declare module "pinia"` code from this plugin did not get included in the library code and TS wasn't aware of the plugin at all. 

Adding
```
import { PluginOptions } from 'pinia-plugin-persistedstate-2'
```

to the file with the store definition (even if I don't actually use the type) makes the compiler error go away and all seems to work, yay!
